h2/src/main/org/h2/expression/Function.java

```diff
@@ -1370,6 +1370,10 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             }
             String regexpMode = v3 == null || v3.getString() == null ? "" :
                     v3.getString();
+            if(database.getMode().getEnum().equals(Mode.ModeEnum.PostgreSQL)) {
+                // PostgreSQL doesn't do global replaces without a 'g' flag, but H2 does
+                regexpMode = regexpMode.replaceAll("g", "");
```

Comment:
From my point of view `makeRegexpFlags()` is a more efficient place to ignore the `g` flag. `String.replaceAll()` is not so fast.

Also you may want to improve PostgreSQL compatibility mode by introduction of conditional usage of `Matcher.replaceFirst()` instead of `Matcher.replaceAll()` a few lines below when H2 is in PostgreSQL compatibility mode and `regexpMode` does not contain the `g` character.

Reply:
Thanks - both great comments. Will amend.